tokio-trace/src/span.rs

```diff
@@ -44,8 +90,98 @@
 //! # }
 //! ```
 //!
+//! **Note:** Since entering a span takes `&self`, and `Span`s are `Clone`,
+//! `Send`, and `Sync`, it is entirely valid for multiple threads to enter the
+//! same span concurrently.
 //!
-//! # The Span Lifecycle
+//! ## Span Relationships
+//!
+//! Spans form a tree structure — unless it is a root span, all spans have a
+//! _parent_, and may have one or more _children_. When a new span is created,
+//! the current span becomes the new span's parent. The total execution time of
```

Comment:
This answers some of the questions from above, but raises a few more:

 - How is the "current" span determined?
 - What if I have entered multiple spans?
 - Is this actually guaranteed, or dependent on the subscriber?